pkg/util/port_split.go

```diff
@@ -38,3 +41,19 @@ func SplitPort(id string) (name, port string, valid bool) {
 	}
 	return id, "", len(id) > 0
 }
+
+// ParsePort parses a raw host string of the form "host:port".
+// Will error if port is absent, non-numeric, or out of allowed port range.
+func ParsePort(rawhost string) (host string, port int, err error) {
```

Comment:
I was pretty lazy about testing for this POC. I'll definitely add them before we're done.